### ISSUE TITLE
Input Date Picker - Ensure dialogs use proper structure - (2036569969)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5582]
### Module: 
 17a Input Date Picker 

### Violation: 
 Ensure dialogs use proper structure 

![image](https://user-images.githubusercontent.com/5023024/202756896-f76e2a67-65dc-4766-85a0-29eb31a0be57.png)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036569969 

### Severity: 
 8 

### Description: 
 [Issue]
The calendar dialog (All) does not have role="dialog" or an accessible name.

[User Impact]
Screen reader users will be unable to determine where the dialog ends and begins. They will also have difficulty determining the purpose of the dialog.

[Code Reference] 
 ```
<calcite-date-picker tabindex="0" role="application" scale="s" calcite-hydrated="">
<div class="header">
     <a aria-disabled="false" aria-label="Previous month" class="chevron" href="#" 
     role="button" tabindex="0">
(...)
      </a>
</div>
</calcite-date-picker>
 ``` 

### Note: 
 [Suggestion]
Ensure dialogs use proper structure. Dialog containers must have role="dialog". If the dialog has a visible title, set aria-labelledby on the dialog container to the ID of the visible title. For dialogs without a visible title, set a descriptive aria-label on the dialog element.

[Compliant Code Example] 
 ```
N/A


 ```
[Recommended Reading]
For more information about creating accessible modal dialogs, please see the W3C Web Accessibility Initiative&#039;s ARIA Authoring Practices Guide: https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/ 

### Media Type: 
 Dialogs 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Since this request would change the role to a dialog, we could also add in keyboard trapping so AT users can cycle through the different options, ensuring they know the full extent of the dialog and what's available to them.

**Comment by geospatialem** [comment]
> Since this request would change the role to a dialog, we could also add in keyboard trapping so AT users can cycle through the different options, ensuring they know the full extent of the dialog and what's available to them.

Added #6668 to address the keyboard trap portion of the request.

**Comment by jcfranco** [comment]
Translation process has been cleared up and is no longer holding this up. This is now waiting on https://github.com/Esri/calcite-components/issues/6773.

**Comment by geospatialem** [comment]
Verified on `1.3.0-next.12` on Mac OS with VoiceOver and on Windows with JAWS and NVDA.
